# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/res/composite.glsl
# Commit: 8d13c0ab66f0
# Full Hash: 8d13c0ab66f0ac137b1461d3451ecb9361ec2d5e
# Author: Jamie Nicol <jnicol@mozilla.com>
# Date: 2023-03-22 21:15:54
# Regressor Bug: 1823411
# File Overlap Count: 1
# Description:
#   Bug 1823411 - Give all webrender's shader varyings an explicit precision. r=gfx-reviewers,lsalzman
#   
#   Mali profiling tools have shown we are frequently fragment bound, in
#   particular due to varying interpolation. To help mitigate this, we
#   should use mediump where possible. Currently most of our varyings are
# ==============================================================================

diff -r 3919fbd972a6 -r 8d13c0ab66f0 gfx/wr/webrender/res/composite.glsl
--- a/gfx/wr/webrender/res/composite.glsl	Tue Mar 21 19:04:22 2023 +0000
+++ b/gfx/wr/webrender/res/composite.glsl	Tue Mar 21 19:05:19 2023 +0000
@@ -16,28 +16,28 @@
 #endif
 
 #ifdef WR_FEATURE_YUV
-YUV_PRECISION flat varying vec3 vYcbcrBias;
-YUV_PRECISION flat varying mat3 vRgbFromDebiasedYcbcr;
+flat varying YUV_PRECISION vec3 vYcbcrBias;
+flat varying YUV_PRECISION mat3 vRgbFromDebiasedYcbcr;
 // YUV format. Packed in to vector to avoid bug 1630356.
-flat varying ivec2 vYuvFormat;
+flat varying mediump ivec2 vYuvFormat;
 
 #ifdef SWGL_DRAW_SPAN
-flat varying int vRescaleFactor;
+flat varying mediump int vRescaleFactor;
 #endif
-varying vec2 vUV_y;
-varying vec2 vUV_u;
-varying vec2 vUV_v;
-flat varying vec4 vUVBounds_y;
-flat varying vec4 vUVBounds_u;
-flat varying vec4 vUVBounds_v;
+varying highp vec2 vUV_y;
+varying highp vec2 vUV_u;
+varying highp vec2 vUV_v;
+flat varying highp vec4 vUVBounds_y;
+flat varying highp vec4 vUVBounds_u;
+flat varying highp vec4 vUVBounds_v;
 #else
-varying vec2 vUv;
+varying highp vec2 vUv;
 #ifndef WR_FEATURE_FAST_PATH
-flat varying vec4 vColor;
-flat varying vec4 vUVBounds;
+flat varying mediump vec4 vColor;
+flat varying highp vec4 vUVBounds;
 #endif
 #ifdef WR_FEATURE_TEXTURE_EXTERNAL_ESSL1
-uniform vec2 uTextureSize;
+uniform mediump vec2 uTextureSize;
 #endif
 #endif
 